app.server/data/catalog/catalog.ts

```diff
@@ -0,0 +1,157 @@
+import * as config from '../../config/config';
+import { GetCollectionsResult, WMSData } from '../../handlers/collections/models';
+import { Query } from '../../../app.client/components/models/Query';
+import { GetProductsResult } from '../../handlers/products/models';
+import axios from 'axios';
+
+import urljoin = require('url-join');
+import turf = require('turf');
+
+export class Catalog {
+  collectionSearchEndpoint: string = 'search/collection';
+  collectionSearchOGCPattern: string = 'scotland-gov/lidar/ogc';
+  collectionSearchLidarPattern: string = 'scotland-gov/lidar/phase*';
+  productSearchEndpoint: string = 'search/product';
+
+  geoserverLayers: { [id: string]: { base_url: string, name: string } };
+
+  private addToGeoserverLayers(name: string, wms: WMSData) {
+    if (this.geoserverLayers) {
+      this.geoserverLayers[name] = wms;
+    } else {
+      this.geoserverLayers = {
+        [name]: wms
+      };
+    }
+  }
+
+  async getOGCServiceList(): Promise<{ [id: string]: WMSData }> {
+    let maxBbox = [-180.0, -85.06, 180.0, 85.06]
+    let layers = await this.getProducts({
+      collections: [this.collectionSearchOGCPattern],
+      offset: 0,
+      limit: 50,
+      bbox: maxBbox,
+      start: '',
+      end: ''
+    }).catch((err) => { console.error(err); throw err });
+
+    layers.collections[0].products.forEach((layer) => {
+      if (layer.data.wms && layer.data.catalog) {
+        this.addToGeoserverLayers(layer.data.catalog.collection, {
+          base_url: layer.data.wms.base_url,
+          name: layer.data.wms.name
+        });
+      }
+    });
+
+    return Promise.resolve(this.geoserverLayers);
+  }
+
+  getOGCServiceForCollection(collection: string): WMSData {
+    return this.geoserverLayers[collection];
+  }
+
+  public async getCollections(pattern: string = this.collectionSearchLidarPattern): Promise<GetCollectionsResult> {
+    return axios.get(urljoin(config.CATALOG_API_URL, this.collectionSearchEndpoint, pattern)).then((response) => {
+      return {
+        collections: (response.data.result).map((collection: any) => {
+          return {
+            id: collection.name,
+            metadata: {
+              title: collection.metadata.title,
+              abstract: collection.metadata.abstract,
+              useConstraints: collection.metadata.useConstraints
+            },
+            metadataExternalLink: collection.metadata.additionalInformationSource,
+            data: {
+              wms: this.getOGCServiceForCollection(collection.name)
+            }
+          }
+        })
+      };
+    });
+  }
+
+  private generateWKTFromBBOX(bbox: number[]): string {
+    // [minX, minY, maxX, maxY]
```

Comment:
Could be a nice spot for some array destructuring, something like

    let [minX, minY, maxX, maxY] = bbox
    return `POLYGON((${minX} ${minY}, ${minX} ${maxY}, ${maxX} ${maxY}, ${maxX} ${minY}, ${minX} ${minY}))`

Reply:
made change

Reply:
Looks great 752d90d 👍